app/javascript/src/Settings/components/Common/SelectGroup.jsx

```diff
@@ -0,0 +1,44 @@
+// @flow
+
+import * as React from 'react'
+import { useState } from 'react'
+import { FormGroup, Select, SelectOption } from '@patternfly/react-core'
+import type { FieldGroupProps, FieldCatalogProps } from 'Settings/types'
+
+type Props = FieldGroupProps & FieldCatalogProps
+
+const SelectGroup = ({
+  label,
+  name,
+  hint,
+  value,
+  catalog
+}: Props) => {
+  const [ selectedValue, setSelectedValue ] = useState(value)
+  const [ isExpanded, setIsExpanded ] = useState(false)
+  const onSelect = (_e, selection) => {
+    setIsExpanded(false)
+    setSelectedValue(selection.key)
+  }
+
+  return (
+    <FormGroup label={label} fieldId={`service_proxy_attributes_${name}_select`} helperText={hint}>
+      <input name={`service[proxy_attributes][${name}]`} type='hidden' value={selectedValue} />
+      <Select
+        selections={catalog[selectedValue]}
+        onToggle={setIsExpanded}
+        onSelect={onSelect}
+        isExpanded={isExpanded}
+        id={`service_proxy_attributes_${name}_select`}
+      >
+        {Object.keys(catalog).map(key => (
```

Comment:
With underscore `map` it would look much better (IMO):
```suggestion
        {map(catalog, (value, key) => <SelectOption key={key} value={{ key, toString: () => value }} />
```

Reply:
I'd rather install Ramda, still is not worth it, it's used just in 2 places. I'd rather do it when we find ourselves using more than one fn.

Reply:
I don't know Ramda but the advantage of underscore is we can install just 1 method, which is almos the same as adding our own helper function.

I still rather have this extracted into a function, even if it's used in 2 places. Also not super important either :P

Reply:
I mean, it's not like we are applying hardcore hacky functions in each place... we are just mapping some Object keys, using built-in prototype, not quite a big deal :)
It was applicable to  https://github.com/3scale/porta/pull/1315#discussion_r341023581, but since it was simplified, I don't think it's worth to install a new dependency or create an util fn just for this. And I know that we could just import the fn, and that we will most probably end up installing either underscore or Ramda... but I don't believe that line justify making it part of this PR.

Reply:
You can import only `map` but anyway you need to add a dependency, I wouldn't do it, at least not here.